core/shared/src/main/scala/internal/JsonLogAppender.scala

```diff
@@ -0,0 +1,59 @@
+package zio.logging.internal
+
+import zio.Cause
+import zio.logging.JsonEscape
+
+private[logging] class JsonLogAppender extends LogAppender {
+  self =>
+  private val sb   = new StringBuilder
+  private var leaf = true
+
+  def appendCause(cause: Cause[Any]): Unit = appendText(cause.prettyPrint)
+
+  override def appendElement(appendValue: LogAppender => Unit): Unit = {
+    val appender    = new JsonLogAppender
+    appendValue(appender)
+    val appendedStr = appender.toString()
+    if (appendedStr.nonEmpty) {
+      if (appender.leaf) {
+        // Leaf nodes should be escaped and quoted
+        sb.append(JsonEscape.jsonEscapedQuoted(appendedStr))
+      } else {
+        sb.append(appendedStr)
+      }
+    }
+
+    self.leaf = false
+  }
+
+  def appendNumeric[A](numeric: A): Unit = appendText(numeric.toString)
+
+  def appendSeparator(): Unit = appendText(",")
+
+  def appendText(text: String): Unit = {
```

Comment:
It could be useful to aggregate the unstructured text. If you are inside the value part of a key/value pair, then when `closeValue()` is called, you collect it and JSON escape it and put the double quotes around it. On the other hand, if you are not inside the value part of a key/value pair, then you can store this into some `unstructuredText` builder that just collects all the unstructured text.